installer/pkg/config-generator/tls.go

```diff
@@ -121,10 +134,104 @@ func (c *ConfigGenerator) GenerateTLSConfig(clusterDir string) error {
 		DNSNames:     []string{baseAddress, fmt.Sprintf("%s.%s", "*", baseAddress)},
 		Subject:      pkix.Name{CommonName: baseAddress, Organization: []string{"ingress"}},
 		Validity:     validityThreeYears,
-		IsCA:         false}
+		IsCA:         false,
+	}
 
 	if _, _, err := generateCert(clusterDir, kubeCAKey, kubeCACert, ingressKeyPath, ingressCertPath, cfg); err != nil {
-		return fmt.Errorf("failed to generate ingress CAs: %v", err)
+		return fmt.Errorf("failed to generate ingress CA: %v", err)
+	}
+
+	// Kube admin certs
+	cfg = &tls.CertCfg{
+		KeyUsages:    x509.KeyUsageKeyEncipherment | x509.KeyUsageDigitalSignature,
+		ExtKeyUsages: []x509.ExtKeyUsage{x509.ExtKeyUsageServerAuth, x509.ExtKeyUsageClientAuth},
+		Subject:      pkix.Name{CommonName: "system:admin", Organization: []string{"system:masters"}},
+		Validity:     validityThreeYears,
+		IsCA:         false,
+	}
+
+	if _, _, err = generateCert(clusterDir, kubeCAKey, kubeCACert, adminKeyPath, adminCertPath, cfg); err != nil {
+		return fmt.Errorf("failed to generate kube admin certificate: %v", err)
+	}
+
+	// Kube API server certs
+	apiServerAddress, err := cidrhost(c.Cluster.Networking.ServiceCIDR, 1)
+	if err != nil {
+		return fmt.Errorf("can't resolve api server host address: %v", err)
+	}
+	cfg = &tls.CertCfg{
+		KeyUsages:    x509.KeyUsageKeyEncipherment | x509.KeyUsageDigitalSignature,
+		ExtKeyUsages: []x509.ExtKeyUsage{x509.ExtKeyUsageServerAuth, x509.ExtKeyUsageClientAuth},
+		Subject:      pkix.Name{CommonName: "kube-apiserver", Organization: []string{"kube-master"}},
+		DNSNames:     []string{fmt.Sprintf("%s-%s.%s", c.Name, "api", c.BaseDomain), "kubernetes", "kubernetes.default", "kubernetes.default.svc", "kubernetes.default.svc.cluster.local"},
```

Comment:
Instead of using `%s` to format the hard-coded `api` string, why not include that in the template?

```go
fmt.Sprintf("%s-api.%s", c.Name, c.BaseDomain)
```

For precedent on this approach, see [here][1], [here][2], and likely other places.  If the pattern sounds appealing, there are some other `fmt.Sprintf` calls in your PR that could also be updated.

[1]: https://github.com/openshift/installer/blob/90239b0debdc0948cb00fb3124650777373e57bf/installer/pkg/config-generator/generator.go#L282
[2]: https://github.com/openshift/installer/blob/90239b0debdc0948cb00fb3124650777373e57bf/installer/pkg/config-generator/ignition.go#L147

Reply:
fair point 👍